### PR TITLE
Remove the multi broker anti-feature

### PR DESCRIPTION
**Pull request by adamralph**

### CODE REVIEW

**Comment by andreasohlund** [comment]
I like your commit msg prefixes. Are you following some kind of standard?

Are you using the metadata in any way?

File: src/NServiceBus.RabbitMQ/Connection/IClusterHostSelectionStrategy.cs [hunk 1]
```diff
@@ -1,42 +0,0 @@
-namespace EasyNetQ
```
Comment (andreasohlund) [inline src/NServiceBus.RabbitMQ/Connection/IClusterHostSelectionStrategy.cs]: We have a praxis of "doing more than semver requires" by leaving any public type in for one extra major to give users a helpful obsolete message. Here is how it's done in the core:

https://github.com/Particular/NServiceBus/blob/develop/src/NServiceBus.Core/v6-obsoletes.cs

Does this make sense?

File: src/NServiceBus.RabbitMQ.Tests/ConnectionString/ConnectionConfigurationTests.cs [hunk 2]
```diff
@@ -11,7 +11,8 @@ public class ConnectionConfigurationTests
         #region Setup/Teardown
```
Comment (andreasohlund) [inline src/NServiceBus.RabbitMQ.Tests/ConnectionString/ConnectionConfigurationTests.cs]: I know you didn't add them, but boo for regions! ;)

**Comment by andreasohlund** [comment]
Nitpick: we tend to name branches with a descriptive name to avoid having to look up what `19` is

Perhaps `remove-multi-broker-feature` ?

or `19-remove-multi-broker-feature`

File: src/NServiceBus.RabbitMQ/Connection/RabbitMqConnectionFactory.cs [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+    class RabbitMqConnectionFactory
+    {
+        public ConnectionConfiguration Configuration { get; private set; }
+
+        public RabbitMqConnectionFactory(ConnectionConfiguration connectionConfiguration)
+        {
+            if (connectionConfiguration == null)
+            {
+                throw new ArgumentNullException("connectionConfiguration");
```
Comment (SimonCropp) [inline src/NServiceBus.RabbitMQ/Connection/RabbitMqConnectionFactory.cs]: why r we arg checking in an internal class? shouldnt this be verified at the point of the public API?

**Comment by adamralph** [comment]
@andreasohlund I'm not following any standard with my commit prefixes, they're just something I made up. I like to isolate my commits to red, green, refactor whenever possible. I don't use the meta data in code, at the moment it's just eyeball consumption.

**Comment by adamralph** [comment]
@andreasohlund fair point about the branche name. `91` is a little lazy. I'll use the `{number}-{description slug}` format in future.

File: src/NServiceBus.RabbitMQ/Connection/IClusterHostSelectionStrategy.cs (hunk 1 above)
Comment (adamralph) [inline src/NServiceBus.RabbitMQ/Connection/IClusterHostSelectionStrategy.cs]: Yep, makes sense. I'll re-instate with obsoletes.

File: src/NServiceBus.RabbitMQ/Connection/RabbitMqConnectionFactory.cs (hunk 3 above)
Comment (adamralph) [inline src/NServiceBus.RabbitMQ/Connection/RabbitMqConnectionFactory.cs]: Fair point. BTW, this class is a rename from `ClusterAwareConnectionFactory` so most of the code, including this check, comes from there. I'll re-visit and see if it makes sense to remove/move it.

File: src/NServiceBus.RabbitMQ.Tests/ConnectionString/ConnectionConfigurationTests.cs (hunk 2 above)
Comment (adamralph) [inline src/NServiceBus.RabbitMQ.Tests/ConnectionString/ConnectionConfigurationTests.cs]: double boo! I'll kill them :wink:

File: src/NServiceBus.RabbitMQ/Connection/RabbitMqConnectionFactory.cs (hunk 3 above)
Comment (SimonCropp) [inline src/NServiceBus.RabbitMQ/Connection/RabbitMqConnectionFactory.cs]: Ok no worries. If this arg checking was preexisting then it doesn't need to be fixed in this pr

Comment (adamralph) [inline src/NServiceBus.RabbitMQ/Connection/RabbitMqConnectionFactory.cs]: OK, I'll leave it for now. I think I've come to a point with this PR that I'm polishing turds which may be removed entirely by a deeper refactoring.

It's sooo tempting to jump down the refactoring rabbit hole in this repo... :stuck_out_tongue:

File: src/NServiceBus.RabbitMQ/Connection/IClusterHostSelectionStrategy.cs (hunk 1 above)
Comment (adamralph) [inline src/NServiceBus.RabbitMQ/Connection/IClusterHostSelectionStrategy.cs]: @andreasohlund I've re-introduced the obsoleted interface in v3-obsoletes.cs. Do you think the PR is good to go?

**Comment by andreasohlund** [comment]
Looks good to go

**Pull request merged**